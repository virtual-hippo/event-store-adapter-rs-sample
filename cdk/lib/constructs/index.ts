export * from "./app-function";
export * from "./cdn";
