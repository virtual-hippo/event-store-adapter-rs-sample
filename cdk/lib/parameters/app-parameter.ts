import * as cdk from "aws-cdk-lib";

export interface AppParameter {
  readonly env: cdk.Environment;
}
