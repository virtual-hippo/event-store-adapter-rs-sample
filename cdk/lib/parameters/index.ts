export * from "./app-parameter";
export * from "./parameters";
