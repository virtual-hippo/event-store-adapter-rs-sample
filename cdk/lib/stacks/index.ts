export * from "./event-store-adapter-rs-sample-stack";
