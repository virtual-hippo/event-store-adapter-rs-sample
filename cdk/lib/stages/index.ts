export * from "./event-store-adapter-rs-sample-stage";
