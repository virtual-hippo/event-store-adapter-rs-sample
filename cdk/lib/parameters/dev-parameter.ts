import { AppParameter } from "./app-parameter";

export const DEV_PARAMETER: AppParameter = {
  env: {
    // 公開リポジトリのためアカウント ID は明示的に記載しない
    // account: "123456789012",
    region: "ap-northeast-1",
  },
} as const;
