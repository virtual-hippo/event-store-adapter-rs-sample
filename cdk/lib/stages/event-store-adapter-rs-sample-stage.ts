import * as cdk from "aws-cdk-lib";
import { Construct } from "constructs";
import { AppParameter, EnvType } from "../parameters";
import { EventStoreAdapterRsSampleStack, GlobalSecurityStack } from "../stacks";

export interface EventStoreAdapterRsSampleStageProps extends cdk.StageProps {
  readonly envType: EnvType;
  readonly appParameter: AppParameter;
}

export class EventStoreAdapterRsSampleStage extends cdk.Stage {
  // for test
  readonly eventStoreAdapterRsSampleStack: EventStoreAdapterRsSampleStack;
  readonly globalSecurityStack: GlobalSecurityStack;

  constructor(scope: Construct, id: string, props: EventStoreAdapterRsSampleStageProps) {
    super(scope, id, props);

    const globalSecurityStack = new GlobalSecurityStack(this, "GlobalSecurity", {
      env: {
        account: props.appParameter.env.account || process.env.CDK_DEFAULT_ACCOUNT,
        region: "us-east-1",
      },
    });

    const eventStoreAdapterRsSampleStack = new EventStoreAdapterRsSampleStack(this, "EventStoreAdapterRsSample", {
      // stack props
      env: {
        account: props.appParameter.env.account || process.env.CDK_DEFAULT_ACCOUNT,
        region: props.appParameter.env.region || process.env.CDK_DEFAULT_ACCOUNT,
      },
      tags: {
        Environment: props.envType,
        SysName: "event-store-adapter-rs-sample",
      },
      // custome props
      envType: props.envType,
      appParameter: props.appParameter,
      contentsHashCalculatorFnArnParameterName: globalSecurityStack.contentsHashCalculatorFnArnParameterName,
    });

    eventStoreAdapterRsSampleStack.addDependency(globalSecurityStack);

    this.globalSecurityStack = globalSecurityStack;
    this.eventStoreAdapterRsSampleStack = eventStoreAdapterRsSampleStack;
  }
}
