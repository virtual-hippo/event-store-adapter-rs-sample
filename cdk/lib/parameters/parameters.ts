import { AppParameter } from "./app-parameter";
import { DEV_PARAMETER } from "./dev-parameter";

export const envs = ["Dev01"] as const;
export type EnvType = (typeof envs)[number];

export const APP_PARAMETERS: Record<EnvType, AppParameter> = {
  Dev01: DEV_PARAMETER,
} as const;
