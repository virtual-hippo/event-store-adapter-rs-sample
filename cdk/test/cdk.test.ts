import * as cdk from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import { APP_PARAMETERS } from "../lib/parameters";
import { EventStoreAdapterRsSampleStage } from "../lib/stages";

export const serializer = {
  test: (val: unknown) => typeof val === "string",
  serialize: (val: string) => {
    return `"${val
      // cloudfront function の簡易認証に使う authString をダミー値に置き換え
      .replace(/const authString = (.+);/, "const authString = REPLACED;")
      // PhysicalResourceId の unix timestamp をダミー値に置き換え
      .replace(/(PhysicalResourceId-[0-9]{13})/, "PhysicalResourceId-REPLACED")
      // Asset hash をダミー値に置き換え
      .replace(/([A-Fa-f0-9]{64}.zip)/, "HASH_REPLACED.zip")
      // Construct address をダミー値に置き換え
      .replace(/[a-f0-9]{42}/, "[CONSTRUCT_ADDR_REPLACED]")}"`;
  },
};

test("Snapshot test", () => {
  const app = new cdk.App();

  const stage = new EventStoreAdapterRsSampleStage(app, "Dev01", {
    envType: "Dev01",
    appParameter: APP_PARAMETERS.Dev01,
  });

  const template = Template.fromStack(stage.eventStoreAdapterRsSampleStack);

  expect.addSnapshotSerializer(serializer);
  expect(template.toJSON()).toMatchSnapshot();
});
