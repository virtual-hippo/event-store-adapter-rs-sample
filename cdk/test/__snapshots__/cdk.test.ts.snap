// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshot test 1`] = `
{
  "Outputs": {
    "AppFunctionWriteApiFunctionUrl1AE33250": {
      "Value": {
        "Fn::GetAtt": [
          "AppFunctionWriteApiFunctionFunctionUrl043725AB",
          "FunctionUrl",
        ],
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "AppFunctionWriteApiFunction3C482938": {
      "DependsOn": [
        "AppFunctionWriteApiFunctionServiceRole89C3ED13",
      ],
      "Properties": {
        "Architectures": [
          "x86_64",
        ],
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-ap-northeast-1",
          },
          "S3Key": "HASH_REPLACED.zip",
        },
        "Environment": {
          "Variables": {
            "APP__API__ALLOW_ORIGINS": "http://localhost:3000,http://localhost:5173,http://localhost:8080",
            "APP__AWS__ACCESS_KEY_ID": "x",
            "APP__AWS__ENDPOINT_URL": "x",
            "APP__AWS__REGION_NAME": "ap-northeast-1",
            "APP__AWS__SECRET_ACCESS_KEY": "x",
            "APP__PERSISTENCE__JOURNAL_AID_INDEX_NAME": "journal-aid-index",
            "APP__PERSISTENCE__JOURNAL_TABLE_NAME": "journal",
            "APP__PERSISTENCE__SHARD_COUNT": "64",
            "APP__PERSISTENCE__SNAPSHOT_AID_INDEX_NAME": "snapshot-aid-index",
            "APP__PERSISTENCE__SNAPSHOT_INTERVAL": "10",
            "APP__PERSISTENCE__SNAPSHOT_TABLE_NAME": "snapshot",
          },
        },
        "Handler": "bootstrap",
        "Role": {
          "Fn::GetAtt": [
            "AppFunctionWriteApiFunctionServiceRole89C3ED13",
            "Arn",
          ],
        },
        "Runtime": "provided.al2023",
      },
      "Type": "AWS::Lambda::Function",
    },
    "AppFunctionWriteApiFunctionFunctionUrl043725AB": {
      "Properties": {
        "AuthType": "AWS_IAM",
        "Cors": {
          "AllowHeaders": [
            "*",
          ],
          "AllowMethods": [
            "POST",
          ],
          "AllowOrigins": [
            "http://localhost:3000",
            "http://localhost:5173",
            "http://localhost:8080",
          ],
          "MaxAge": 300,
        },
        "TargetFunctionArn": {
          "Fn::GetAtt": [
            "AppFunctionWriteApiFunction3C482938",
            "Arn",
          ],
        },
      },
      "Type": "AWS::Lambda::Url",
    },
    "AppFunctionWriteApiFunctionServiceRole89C3ED13": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
